src/content-helper/dashboard-widget/components/top-posts-list-item.tsx

```diff
@@ -104,13 +108,54 @@ function getPostThumbnailElement( { post }: TopPostListItemProps ): JSX.Element
 	);
 }
 
+/**
+ * Returns a span element with the desired metric data for the Post.
+ *
+ * Currently, only the `views` and `avg_engaged` metrics are supported.
+ *
+ * @since 3.10.0
+ *
+ * @param {TopPostListItemProps} props The component's props.
+ *
+ * @return {JSX.Element} The resulting JSX Element.
+ */
+function PostListItemMetric( { metric, post }: TopPostListItemProps ): JSX.Element {
+	if ( 'views' === metric ) {
+		return (
+			<span className="parsely-top-post-metric-data">
+				<span className="screen-reader-text">
+					{ __( 'Number of Views', 'wp-parsely' ) }
+				</span>
+				{ formatToImpreciseNumber( post.views.toString() ) }
+			</span>
+		);
+	}
+
+	if ( 'avg_engaged' === metric ) {
+		return (
+			<span className="parsely-top-post-metric-data">
+				<span className="screen-reader-text">
+					{ __( 'Average Time', 'wp-parsely' ) }
+				</span>
+				{ post.avgEngaged }
+			</span>
+		);
+	}
+
+	return (
+		<span className="parsely-top-post-metric-data">
+			{ __( 'Unsupported metric specified.', 'wp-parsely' ) }
```

Comment:
Isn't it too long .... how about a simple `-`?

Reply:
You're right, this isn't the whole row we're populating.

Reply:
Noting that when we've got a case like that, the API shows an error anyway and nothing is displayed.